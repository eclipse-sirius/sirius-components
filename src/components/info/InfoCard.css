/*******************************************************************************
 * Copyright (c) 2018 Obeo and others.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2
 * which accompanies this distribution and is available at
 * https://www.eclipse.org/legal/epl-2.0.
 *******************************************************************************/

.infocard {
  display: grid;
  grid-template-rows: min-content min-content;
  grid-row-gap: var(--layoutDimension-m);
  padding: var(--layoutDimension-l);
}

.infocard-title {
  text-align: center;
  font-size: var(--fontSize-xxxl);
  font-weight: var(--fontWeight-bold);
  line-height: var(--lineHeight-xxxl);
}

.infocard-message {
  text-align: center;
  font-size: var(--fontSize-l);
  line-height: var(--lineHeight-l);
}
