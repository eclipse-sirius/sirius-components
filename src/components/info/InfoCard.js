/*******************************************************************************
 * Copyright (c) 2018 Obeo and others.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2
 * which accompanies this distribution and is available at
 * https://www.eclipse.org/legal/epl-2.0.
 *******************************************************************************/
import React from 'react';
import PropTypes from 'prop-types';

import { classNames } from '../../common/classnames';

import { Card } from '../cards/Card';
import { Spacing } from '../spacing/Spacing';
import { L } from '../spacing/SpacingConstants';

import './InfoCard.css';

const propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string
};

const INFOCARD__CLASS_NAMES = 'infocard';
const INFOCARD_CONTAINER__CLASS_NAMES = 'infocard-container';
const INFOCARD_TITLE__CLASS_NAMES = 'infocard-title';
const INFOCARD_MESSAGE__CLASS_NAMES = 'infocard-message';

/**
 * The InfoCard component is used to display some information with a catchy card.
 */
export const InfoCard = ({ className, title, message, ...props }) => {
  const infoCardClassNames = classNames(INFOCARD__CLASS_NAMES, className);
  return (
    <Card {...props} className={infoCardClassNames}>
      <Spacing top={L} right={L} bottom={L} left={L}>
        <div className={INFOCARD_CONTAINER__CLASS_NAMES}>
          <h1 className={INFOCARD_TITLE__CLASS_NAMES}>{title}</h1>
          <p className={INFOCARD_MESSAGE__CLASS_NAMES}>{message}</p>
        </div>
      </Spacing>
    </Card>
  );
};
InfoCard.propTypes = propTypes;
