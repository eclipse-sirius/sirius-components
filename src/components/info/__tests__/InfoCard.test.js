/*******************************************************************************
 * Copyright (c) 2018 Obeo and others.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2
 * which accompanies this distribution and is available at
 * https://www.eclipse.org/legal/epl-2.0.
 *******************************************************************************/

import React from 'react';
import Renderer from 'react-test-renderer';

import { InfoCard } from '../InfoCard';

describe('InfoCard', () => {
  it('renders an info card', () => {
    const infoCard = Renderer.create(<InfoCard title="42" message="Projects" />);
    expect(infoCard.toJSON()).toMatchSnapshot();
  });
});
