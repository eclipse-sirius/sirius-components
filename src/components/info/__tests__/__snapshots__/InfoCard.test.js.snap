// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard renders an info card 1`] = `
<div
  className="card infocard"
>
  <div
    className="css-1nc71gt"
  >
    <div
      className="infocard-container"
    >
      <h1
        className="infocard-title"
      >
        42
      </h1>
      <p
        className="infocard-message"
      >
        Projects
      </p>
    </div>
  </div>
</div>
`;
